src/Dialect/ONNX/Rewrite.cpp

```diff
@@ -132,6 +132,395 @@ DenseElementsAttr createDenseElementsAttrFromShapeOp(
+// The results are used weight tensor for Conv op.
+Value reverseWeightTensor(
+    PatternRewriter &rewriter, Location loc, Value input) {
+  ShapedType inputType = input.getType().cast<ShapedType>();
+  Type elementType = inputType.getElementType();
+  assert(inputType.hasRank() && "Need rank to reverse weight tensor.");
+  // Reverse elements using ReverseSequence op.
+  // ReverseSequence op can reverse elements in the first and second dimensions.
+  // So, transpose target dimension in advance.
+  // 1. Transpose NxCxD0xD1xD2x... to D0xD1xD2x ... xNxC
+  //  perms for 3D weight(ConvTranspose1D) [2, 0, 1]
+  //  perms for 4D weight(ConvTranspose2D) [2, 3, 0, 1]
+  //  perms for 5D weight(ConvTranspose3D) [2, 3, 4, 0, 1]
+  int64_t spatialOffset = 2; // for N and C
+  int64_t spatialRank = inputType.getRank() - spatialOffset;
+  SmallVector<int64_t, 4> permsval;
+  for (int i = 0; i < spatialRank; ++i)
+    permsval.emplace_back(spatialOffset + i);
+  for (int i = 0; i < spatialOffset; ++i)
+    permsval.emplace_back(i);
+  ArrayRef<int64_t> perms(permsval);
+  Value transposedInput = emitONNXTranspose(loc, rewriter, input, perms);
+  // 2. Reverse all elements for spatial dimension
+  // 2.1 The first two dimensions can be reversed using two ReverseSequence in
+  // the first loop. Reverse D0 and D1 , then D2 and D3 ... for "D0 x D1 x D2 x
+  // ... x Dn x N x C"
+  for (int i = 0; i < spatialRank / 2; i += 2) {
+    // Reverse the first and sconde dimension
+    Value reverse0 =
+        reverseAllElements(rewriter, loc, transposedInput, /*dimension*/ 0);
+    Value reverse1 =
+        reverseAllElements(rewriter, loc, reverse0, /*dimension*/ 1);
+    // Move reversed two dimensions to the last for next reverse.
+    SmallVector<int64_t, 4> permsval0;
+    for (int j = 0; j < inputType.getRank() - 2; ++j)
+      permsval0.emplace_back(j + 2);
+    for (int j = 0; j < 2; ++j)
+      permsval0.emplace_back(j);
+    ArrayRef<int64_t> perms(permsval0);
+    transposedInput = emitONNXTranspose(loc, rewriter, reverse1, permsval0);
+  }
+  // 2.2 The rest of dimension are reversed using single ReverseSequence
+  // Reverse last dimension Dn for "D0 x D1 x D2 x ... x Dn x N x C".
+  // This is done only when rank is odd.
+  for (int i = (spatialRank / 2) * 2; i < spatialRank; ++i) {
+    ShapedType tInType = transposedInput.getType().cast<ShapedType>();
+    ArrayRef<int64_t> tInShape = tInType.getShape();
+    Value reverse0;
+    if (tInShape[1] == ShapedType::kDynamic) {
+      // When N is unknown dim,
+      // reshape "Dn x N x C x D0 xD1 x D2 x ... x Dn-1"
+      // to "Dn x 1 x N x C x D0 xD1 x D2 x ... x Dn-1",
+      // then, reshape back to original shape after reversed.
+      int64_t tInRank = tInShape.size();
+      Type tInShapeType =
+          RankedTensorType::get({tInRank}, rewriter.getI64Type());
+      Value tInShapeVals =
+          rewriter.create<ONNXShapeOp>(loc, tInShapeType, transposedInput);
```

Comment:
Please use OnnxBuilder for this, https://github.com/onnx/onnx-mlir/blob/main/src/Dialect/ONNX/DialectBuilder.hpp#L87.

Reply:
Done for existing ONNXBuilder. Thanks!